main.go

```diff
@@ -165,20 +351,29 @@ func (lb *loadBalancer) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	p = trimPrefixSuffixSlash(p)
 
 	var proxy *httputil.ReverseProxy
-
-	if strings.Contains(p, "/") {
+	var bkIdx int
+	if strings.HasPrefix(p, "prometheus/") {
```

Comment:
Let's not have this type of code @poornas - let's have a proper registered handler which is routed to automatically, and rest can be routed differently.

Reply:
We need to have a correct endpoint for this currently this would look like any URL with `prometheus` will work. This would conflict with the bucket name which is `prometheus` so should add this as a private endpoint somehow with like `/.prometheus/metrics`